### PR TITLE
Add new Redfish facts module to handle Dell-specific OEM extensions

### PR DESCRIPTION
**Pull request by jose-delarosa**
- Extends the existing RedfishUtils class and leverages its methods

##### SUMMARY

Implements code taken out by #51898 and provides module to implement retrieval of all iDRAC-specific configuration settings.

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

idrac_redfish_facts

##### ADDITIONAL INFORMATION

This new module can be used as a model for other OEMs to implement their own Redfish extensions.

### CODE REVIEW

**Review by billdodd — COMMENTED** [review]
Since the 'timeout' PR has now been merged, this PR will need to be updated to add the timeout support. See the recent timeout related commit to idrac_redfish_command.py.

**Review by billdodd — APPROVED** [review]
Looks good.

shipit

**Comment by jose-delarosa** [comment]
shipit

File: lib/ansible/modules/remote_management/redfish/idrac_redfish_facts.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+DOCUMENTATION = '''
+---
+module: idrac_redfish_facts
+version_added: "2.8"
+short_description: Manages servers through iDRAC using Dell Redfish APIs
+description:
+  - Builds Redfish URIs locally and sends them to remote iDRAC controllers to
+    get information back.
+  - For use with Dell iDRAC operations that require Redfish OEM extensions
+options:
```
Comment (gundalow) [inline lib/ansible/modules/remote_management/redfish/idrac_redfish_facts.py:23]: In a follow up PR could you please ensure all `remote_management/redfish` modules have `type:` set for the options?
A single PR which updates all modules would help improve the online documentation

File: lib/ansible/modules/remote_management/redfish/idrac_redfish_facts.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+  password:
+    required: true
+    description:
+      - Password for authentication with iDRAC controller
+  timeout:
+    description:
+      - Timeout in seconds for URL requests to OOB controller
+    default: 10
+    type: int
+    version_added: "2.8"
```
Comment (gundalow) [inline lib/ansible/modules/remote_management/redfish/idrac_redfish_facts.py]:
```suggestion
```

File: lib/ansible/modules/remote_management/redfish/idrac_redfish_facts.py (hunk 1 above)
Comment (jose-delarosa) [reply lib/ansible/modules/remote_management/redfish/idrac_redfish_facts.py:23]: Will do. Once all modules are merged I can submit one PR to update all modules in `remote_management/redfish`. Thanks.

Comment (billdodd) [reply lib/ansible/modules/remote_management/redfish/idrac_redfish_facts.py:23]: Thanks, @gundalow and @jose-delarosa. Good idea. FYI - I created issue #54688 for this.

**Comment by billdodd** [comment]
shipit

**Pull request merged**